File: src/CoMute/Content/js/carpool.js

$(function () {

    $.ajax({
        url: '/api/carpool',
        type: 'get',
        headers: headers,
        success: function (data) {
            $('#item_body').html('');
            if (data.length == 0) $('#item_body').append(`<h2><span class="text-danger"> No car pools created yet!</span></h2>`);
            else data.forEach(AddCarpoo);
        }
    });

});

function AddCarpoo(carpool) {
    console.log(carpool);
    $('#item_body').append(`            <hr />
            <div class="row">
                <div class="col-md-2">${carpool.Origin} </div>
                <div class="col-md-2">${carpool.Destination}</div>
                <div class="col-md-1">${getTime(carpool.DepartureTime)}</div >
                <div class="col-md-1">${getTime(carpool.ExpectArivalTime)}</div>
                <div class="col-md-1">${carpool.AvailableSeats}</div>
                <div class="col-md-1">${carpool.DaysAvailable}</div>
                <div class="col-md-2"><small>${carpool.Notes}</small></div>
                <div class="col-md-2">${carpool.PoolCreationDate}</div>
            </div>`);
   
}

function getTime(datetime) {
    let date = new Date(datetime);
    return date.toLocaleTimeString();
}

File: src/CoMute/Models/Carpool.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CoMute.Web.Models
{
    public class Carpool
    {
        public string departureTime { get; set; }
        public string expectedArrivalTime { get; set; }
        public string origin { get; set; }
        public string daysAvailable { get; set; }
        public string destination { get; set; }
        public string availableSeats { get; set; }
        public string ownerLeader { get; set; }
        public string notes { get; set; }
    }
}
